src/Grid/grid.jl

```diff
@@ -708,11 +708,103 @@ function _addset!(grid::AbstractGrid, name::String, f::Function, _ftype::Functio
     grid
 end
 
+
+"""
+    getfaceedges(face::NTuple{n,Int64})
+
+Returns the edges in a given face.
+
+```julia
+getfaceedges(face)
+```
+"""
+getfaceedges(grid::AbstractGrid, parent::FaceIndex, f::Function; all::Bool=true) =
+    _getentities(grid, parent, f, Ferrite.edges,
+     face -> ntuple(i -> length(face)>2&&i!=length(face) ? (face[i+1],face[i]) : (face[1],face[length(face)]),length(face)), 
+     Ferrite.faces, EdgeIndex; all = all)
+
+
+"""
+    getfacevertices(face::NTuple{n,Int64})
+
+Returns the vertices in a given face.
+
+```julia
+getfacevertices(face)
+```
+"""
+
+getfacevertices(grid::AbstractGrid, parent::FaceIndex, f::Function; all::Bool=true) =
+    _getentities(grid, parent, f, Ferrite.vertices, face -> face,Ferrite.faces, VertexIndex; all = all)
+"""
+    getedgevertices(edge::NTuple{2,Int64})
+
+Returns the vertices in a given edge.
+
+```julia
+getedgevertices(edge)
+```
+"""
+
+
+getedgevertices(grid::AbstractGrid, parent::EdgeIndex, f::Function; all::Bool=true) =
+    _getentities(grid, parent, f, Ferrite.vertices, edge -> edge,Ferrite.edges, EdgeIndex; all = all)
+#Just for validating that the function f applies to the face nodes
+_checkface(grid::AbstractGrid, parent::FaceIndex, f::Function; all::Bool=true) = 
+    _getentities(grid, parent, f, Ferrite.faces, face -> (face,),Ferrite.faces, FaceIndex; all = all)
+function _getentities(grid::AbstractGrid, parent::BoundaryIndex, f::Function, _ftype::Function, _fentities::Function, _fparent::Function, _indextype::Type; all::Bool=true)
+    _set = Set{_indextype}()
+    cells = getcells(grid)
+    parent_cell = cells[parent[1]]
+    entities = _fentities(_fparent(parent_cell)[parent[2]])#_fparent returns global vertices ids of face/edge. _fentities transforms them to pairs for edges.
+    for (cell_idx, cell) in enumerate(cells)
+        cell_entities = _ftype(cell)
+        for entity_idx in findall(x->x ∈ entities, cell_entities) # entities can be edges/vertices in the face/edge
+            pass = all
+                for node_idx in cell_entities[entity_idx] # using cell entities common with boundary face
+                    v = f(grid.nodes[node_idx].x)
+                    all ? (!v && (pass = false; break)) : (v && (pass = true; break))
+                end
+            pass && push!(_set, _indextype(cell_idx, entity_idx))
+        end
+    end
```

Comment:
I think this should be moved to `_addset!`. The reason is that we will merge some topologies which allow more efficient queries on face and edge neighbors (which do not require looping over all cells).